toolbox/library/src/YarpRpc.cpp

```diff
@@ -0,0 +1,257 @@
+    RisingEdgeTrigger,
+    PortName,
+    RpcCommand,
+};
+
+enum InputIndex
+{
+    TriggerSignal = 0,
+};
+
+// BLOCK PIMPL
+// ===========
+
+class YarpRpc::impl
+{
+public:
+    double timeout = 0.25;
+    std::string serverPortName;
+    std::string clientPortName;
+
+    std::unique_ptr<yarp::os::Network> network = nullptr;
+    bool previousTrigger = false;
+    bool risingEdgeTrigger;
+    yarp::os::RpcClient rpcClientPort;
+    yarp::os::Bottle rpcCommand;
+};
+
+// BLOCK CLASS
+// ===========
+
+YarpRpc::YarpRpc()
+    : pImpl{new impl()}
+{}
+
+YarpRpc::~YarpRpc() = default;
+
+unsigned YarpRpc::numberOfParameters()
+{
+    return Block::numberOfParameters() + 3;
+}
+
+bool YarpRpc::parseParameters(BlockInformation* blockInfo)
+{
+    const std::vector<ParameterMetadata> metadata{
+        {ParameterType::BOOL, ParamIndex::RisingEdgeTrigger, 1, 1, "RisingEdgeTrigger"},
+        {ParameterType::STRING, ParamIndex::PortName, 1, 1, "PortName"},
+        {ParameterType::STRING, ParamIndex::RpcCommand, 1, 1, "RpcCommand"}
+    };
+
+    for (const auto& md : metadata) {
+        if (!blockInfo->addParameterMetadata(md)) {
+            bfError << "Failed to store parameter metadata";
+            return false;
+        }
+    }
+
+    return blockInfo->parseParameters(m_parameters);
+}
+
+bool YarpRpc::configureSizeAndPorts(BlockInformation* blockInfo)
+{
+    // INPUT
+    // =====
+    //
+    // 1) Trigger signal
+    //
+    // OUTPUT
+    // ======
+    //
+    // No outputs
+    //
+
+    const bool ok = blockInfo->setPortsInfo(
+        {
+            // Inputs
+            {InputIndex::TriggerSignal, Port::Dimensions{1}, Port::DataType::DOUBLE},
+        },
+        {
+            // Outputs
+        });
+
+    if (!ok) {
+        bfError << "Failed to configure input / output ports.";
+        return false;
+    }
+
+    return true;
+}
+
+bool YarpRpc::initialize(BlockInformation* blockInfo)
+{
+    if (!Block::initialize(blockInfo)) {
+        return false;
+    }
+
+    // PARAMETERS
+    // ==========
+
+    if (!YarpRpc::parseParameters(blockInfo)) {
+        bfError << "Failed to parse parameters.";
+        return false;
+    }
+
+    // Temp variable to store rpc command string
+    std::string command;
+
+    bool ok = true;
+    ok = ok && m_parameters.getParameter("RisingEdgeTrigger", pImpl->risingEdgeTrigger);
+    ok = ok && m_parameters.getParameter("PortName", pImpl->serverPortName);
+    ok = ok && m_parameters.getParameter("RpcCommand", command);
+
+    if (!ok) {
+        bfError << "Failed to read input parameters.";
+        return false;
+    }
+
+    // CLASS INITIALIZATION
+    // ====================
+
+    pImpl->network = std::make_unique<yarp::os::Network>();
+
+    if (!yarp::os::Network::initialized() || !yarp::os::Network::checkNetwork(5.0)) {
+        bfError << "YARP server wasn't found active.";
+        return false;
+    }
+
+    // Open a rpc client port
+    pImpl->clientPortName = "...";
```

Comment:
This variable is not meaningful and it is also dangerous since it is not the real client port name. Can you move the string directly to the `open` method of the port?

Reply:
Done